### DISCUSSION TITLE
How to intercept admin rest api requests to approve or reject it?

### DISCUSSION THREAD

**Discussion by mkopeyka** [keycloak/keycloak#20108] [Q&A]
We have an issue with standard rest api endpoint to update user
    
    PUT /admin/realms/test-realm/users/xxxxxxx-xxxxxxx-xxxxxxxx-xxxxxxxxx
    
Some microservice updates data incorrectly and we cannot find which one.
Is it possible to create some addon for keycloak to "review" request body before applying this change to keycloak entity? We would like to approve or reject changes according to specific rules.

**Comment by danielFesenmeyer** [comment]
I would suggest to implement a ContainerRequestFilter, marked with the annotations `@Provider` and `@PreMatching`, and deploying it as a Keycloak extension. Note that this filter is applied before authentication, but probably that's ok for your use case.

See this tutorial for a simple example: https://www.baeldung.com/jersey-filters-interceptors#1-implementing-a-request-server-filter
